Quadriga/src/main/java/edu/asu/spring/quadriga/rest/TextUploadRestController.java

```diff
@@ -0,0 +1,134 @@
+
+    /**
+     * Controller method for handling Rest requests for Text Uploads
+     * @param wsId
+     *            WorkspaceId for the Textfile as a path variable.
+     * @param projId
+     *            Project for the Textfile as a path variable.
+     * @param response
+     *            Generic HTTP Response Object for returning the XML response
+     * @param request
+     *            Generic HTTP Request Object for RECEIVING the XML response
+     * @param xml
+     *            XML content as a string
+     * @return Returns a Response Entity that contains with the information
+     *         about the stored Textfile
+     * @throws RestException
+     */
+    @RequestMapping(value = "rest/project/{projectid}/workspace/{workspaceid}/text", method = RequestMethod.POST, produces = "application/xml")
+    public ResponseEntity<String> uploadText(@PathVariable("workspaceid") String wsId,
+            @PathVariable("projectid") String projId, HttpServletResponse response, HttpServletRequest request,
+            @RequestBody String xml) throws RestException {
+        String textURI = messages.getProperty("textfiles_location.url");
+        ITextFile txtFile = null;
+        try {
+            txtFile = txtXMLParser.parseTextXML(xml, wsId, projId);
+        } catch (TextFileParseException tfe) {
+            logger.error("Error in Text Rest Controller:", tfe);
+            String errorMsg = errorMessageRest.getErrorMsg(tfe.getMessage());
+            return new ResponseEntity<String>(errorMsg, HttpStatus.BAD_REQUEST);
+        }
+        try {
+            tfManager.saveTextFile(txtFile);
+        } catch (QuadrigaStorageException e) {
+            logger.error("Error in Text Rest Controller:", e);
+            String errorMsg = errorMessageRest.getErrorMsg(e.getMessage());
+            return new ResponseEntity<String>(errorMsg, HttpStatus.INTERNAL_SERVER_ERROR);
+        } catch (FileStorageException e) {
+            logger.error("Error in Text Rest Controller:", e);
+            String errorMsg = errorMessageRest.getErrorMsg(e.getMessage());
+            return new ResponseEntity<String>(errorMsg, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        VelocityEngine engine = null;
+        try {
+            engine = restVelocityFactory.getVelocityEngine(request);
+            engine.init();
+            Template template = engine.getTemplate("velocitytemplates/textfile.vm");
+            VelocityContext context = new VelocityContext(restVelocityFactory.getVelocityContext());
+            context.put("textid", txtFile.getTextId());
+            context.put("refid", txtFile.getRefId());
+            context.put("filename", txtFile.getFileName());
+            context.put("wsid", txtFile.getWorkspaceId());
+            context.put("projid", txtFile.getProjectId());
+            context.put("texturi", textURI + "/" + txtFile.getTextId());
```

Comment:
The URI here I am setting is for the public pages/ texts[quadriga.io/text/]. Should this be part of the TextFile ?